finality-aleph/src/metrics/finality_rate.rs

```diff
@@ -0,0 +1,191 @@
+use std::num::NonZeroUsize;
+
+use lru::LruCache;
+use parking_lot::Mutex;
+use primitives::{BlockHash, BlockNumber};
+use sc_service::Arc;
+use substrate_prometheus_endpoint::{register, Counter, PrometheusError, Registry, U64};
+
+use super::Checkpoint;
+
+const MAX_CACHE_SIZE: usize = 1800;
+const MAX_INNER_SIZE: usize = 64;
+
+#[derive(Clone)]
+pub enum FinalityRateMetrics {
+    Prometheus {
+        own_finalized: Counter<U64>,
+        own_hopeless: Counter<U64>,
+        imported_cache: Arc<Mutex<LruCache<BlockNumber, LruCache<BlockHash, ()>>>>,
```

Comment:
1. add type alias
```
type ImportedHashesCache = Arc<Mutex<LruCache<BlockNumber, LruCache<BlockHash, ()>>>>;
```
2. Why lru instead of bounded vec or just set? this seems strange

Reply:
* Right, Changed to `BoundedVec`.
* Also added the suggested type alias